src/resources/License.py

```diff
@@ -0,0 +1,328 @@
+logger = logging.getLogger(__name__)
+
+class License:
+    def __init__(self, web3: Web3, account, chain_id):
+        self.web3 = web3
+        self.account = account
+        self.chain_id = chain_id
+
+        self.license_template_client = PILicenseTemplateClient(web3)
+        self.license_registry_client = LicenseRegistryClient(web3)
+        self.licensing_module_client  = LicensingModuleClient(web3)
+        self.ip_asset_registry_client = IPAssetRegistryClient(web3)
+
+    def _get_license_terms_id(self, license_terms):
+        logger.info(f"Getting license terms ID for: {license_terms}")
+        return self.license_template_client.getLicenseTermsId(license_terms)
+
+    def registerNonComSocialRemixingPIL(self):
+        try:
+            # Get the license terms for non-commercial social remixing PIL
+            license_terms = get_license_term_by_type(PIL_TYPE['NON_COMMERCIAL_REMIX'])
+            logger.info(f"License terms: {license_terms}")
+
+            # Check if the license terms are already registered
+            license_terms_id = self._get_license_terms_id(license_terms)
+            logger.info(f"License terms ID: {license_terms_id}")
+            if (license_terms_id is not None) and (license_terms_id != 0):
+                return {'licenseTermsId': license_terms_id}
+
+            # Build the transaction
+            transaction = self.license_template_client.build_registerLicenseTerms_transaction(
+                license_terms, {
+                    'from': self.account.address,
+                    'nonce': self.web3.eth.get_transaction_count(self.account.address),
+                    'gas': 2000000,
+                    'gasPrice': self.web3.to_wei('100', 'gwei')  # Adjusted gas price for faster processing
+                }
+            )
+            logger.info(f"Transaction: {transaction}")
+
+            # Sign the transaction using the account object
+            signed_txn = self.account.sign_transaction(transaction)
+            logger.info(f"Signed transaction: {signed_txn}")
+
+            # Send the transaction
+            tx_hash = self.web3.eth.send_raw_transaction(signed_txn.rawTransaction)
+            logger.info(f"Transaction hash: {tx_hash.hex()}")
+
+            # Wait for transaction receipt with a longer timeout
+
+            tx_receipt = self.web3.eth.wait_for_transaction_receipt(tx_hash, timeout=600)  # 10 minutes timeout
+            logger.info(f"Transaction receipt: {tx_receipt}")
+
+            # Parse the event logs for LicenseTermsRegistered
+            target_logs = self._parse_tx_license_terms_registered_event(tx_receipt)
+            return {
+                'txHash': tx_hash.hex(),
+                'licenseTermsId': target_logs
+            }
+
+        except Exception as e:
+            logger.error(f"Error interacting with contract: {e}")
+            raise e
+        
+    def registerCommercialUsePIL(self, minting_fee, currency, royalty_policy, tx_options=None):
```

Comment:
should we have a default value for currency?

Reply:
I don't think so since you can specify currency in license terms, and this generates the license term id since people way want to use different currency. PRobably will mainly use ERC20 but should want to have the option available to switch out to different currencies